Release notes `v0.24.0` (#2701)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

    >>> @qml.qnode(dev)
    ... def circuit(x):
    ...     qml.IsingXX(x, wires=[0,1])
    ...     return qml.mutual_info(wires=[0], wires1=[1])
    >>> circuit(np.pi/2)
    tensor(1.38629436, requires_grad=True)
    ```

  As well as new transforms available in the `qml.qinfo` module,

  - Classical and quantum fisher information transforms:
  
    ```python3
    dev = qml.device("default.qubit", wires=3)

    @qml.qnode(dev)
    def circ(params):
        qml.RY(params[0], wires=1)
        qml.CNOT(wires=(1,0))
        qml.RY(params[1], wires=1)
        qml.RZ(params[2], wires=1)
        return qml.expval(qml.PauliX(0) @ qml.PauliX(1) - 0.5 * qml.PauliZ(1))

    params = np.array([0.5, 1., 0.2], requires_grad=True)
    cfim = qml.qinfo.classical_fisher(circ)(params)
    qfim = qml.qinfo.quantum_fisher(circ)(params)
    ```

    These quantities are typically employed in variational optimization schemes to tilt the gradient in a more favourable direction (see [2103.15191](https://arxiv.org/abs/2103.15191) and [1909.02108](https://arxiv.org/abs/1909.02108)) --- producing what is known as the *natural* gradient. For example:

    ```pycon

>     >>> grad = qml.grad(circ)(params)

Reviewer: It used to be here, although these days I am very cognizant of ensuring our examples are as concise as possible 😆 Since the main purpose here is to show how the CFIM may be used to construct the natural gradient, I think comparing to the actual gradient is not necessary.

  ```python
  dev = qml.device('default.qubit', wires=2)
  x = jnp.array(0.543)
  y = jnp.array(-0.654)

  @jax.jit
  @qml.qnode(dev, diff_method="parameter-shift", interface="jax")
  def circuit(x, y):
      qml.RX(x, wires=[0])
      qml.RY(y, wires=[1])
      qml.CNOT(wires=[0, 1])
      return qml.probs(wires=[1])
  ```
  ```pycon
  >>> circuit(x, y)
  DeviceArray([0.8397495 , 0.16025047], dtype=float32)
  ```

  Note that computing the jacobian of vector-valued QNode is not supported with JAX
  JIT. The output of vector-valued QNodes can, however, be used in
  the definition of scalar-valued cost functions whose gradients can be
  computed.

  For example, one can define a cost function that outputs the first element of
  the probability vector:

  ```python
  def cost(x, y):
      return circuit(x, y)[0]
  ```
  
  ```pycon
  >>> jax.grad(cost, argnums=[0])(x, y)
  (DeviceArray(-0.2050439, dtype=float32),)
  ```

<h4>More drawing styles 🎨</h4>

* New `solarized_light` and `solarized_dark` styles are available for drawing circuit diagram graphics. 
  [(#2662)](https://github.com/PennyLaneAI/pennylane/pull/2662)

<h4>New operations & transforms 🤖</h4>  
  

> * The `IsingXY` gate is now available (see [1912.04424](https://arxiv.org/abs/1912.04424)). 

Reviewer: ```suggestion
* The `qml.IsingXY` gate is now available (see [1912.04424](https://arxiv.org/abs/1912.04424)). 
```

  [(#2649)](https://github.com/PennyLaneAI/pennylane/pull/2649)

> * The `qml.ECR` (echoed cross-resonance) operation is now available (see

Reviewer: Oh in that case, we should instead update `IsingXY` to be `qml.IsingXY`. I think it's useful to readers to indicate where in PL this lives, so they don't have to search for it

  observables.
  [(#2587)](https://github.com/PennyLaneAI/pennylane/pull/2587)

  ```python
  dev = qml.device("default.qubit", wires=1)

  @qml.transforms.split_non_commuting
  @qml.qnode(dev)
  def circuit(x):
      qml.RX(x,wires=0)
      return [qml.expval(qml.PauliX(0)), qml.expval(qml.PauliZ(0))]
  ```

  ```pycon
  >>> print(qml.draw(circuit)(0.5))
  0: ──RX(0.50)─┤  <X>
  \
  0: ──RX(0.50)─┤  <Z>
  ```

<h3>Improvements</h3>

* Expectation values of multiple non-commuting observables from within a single QNode are now
  supported.
  [(#2587)](https://github.com/PennyLaneAI/pennylane/pull/2587)

  ```
  >>> dev = qml.device('default.qubit', wires=1)
  >>> @qml.qnode(dev)
  ... def circuit_rx(x, z):
  ...     qml.RX(x, wires=0)
  ...     qml.RZ(z, wires=0)
  ...     return qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))
  >>> circuit_rx(0.1, 0.3)
  tensor([ 0.02950279, -0.09537451], requires_grad=True)
  ```

* Selecting which parts of parameter-shift Hessians are computed is now possible.
  [(#2538)](https://github.com/PennyLaneAI/pennylane/pull/2538)

  The `argnum` keyword argument for `qml.gradients.param_shift_hessian`

>   is now allowed to be a twodimensional Boolean `array_like`.

Author: ```suggestion
  is now allowed to be a two-dimensional Boolean `array_like`.
```

  CNOT(wires=[0, 1])
  ```

* The `qml.taper` function can now be used to consistently taper any additional observables such as
  dipole moment, particle number, and spin operators using the symmetries obtained from the
  Hamiltonian.
  [(#2510)](https://github.com/PennyLaneAI/pennylane/pull/2510)

> * Sparse Hamiltonians' representation has changed from COOrdinate (COO) to Compressed Sparse Row

Author: ```suggestion
* Sparse Hamiltonians' representation has changed from Coordinate (COO) to Compressed Sparse Row
```

  (CSR) format.
  [(#2561)](https://github.com/PennyLaneAI/pennylane/pull/2561)

  The CSR representation is more performant for arithmetic operations and matrix-vector products.
  This change decreases the `expval()` calculation time for `qml.SparseHamiltonian`, specially for

>   large workflows. In addition, the CRS format consumes less memory for `qml.SparseHamiltonian`

Author: ```suggestion
  large workflows. In addition, the CSR format consumes less memory for `qml.SparseHamiltonian`
```

  In addition, [code examples](https://pennylane.readthedocs.io/en/stable/introduction/unsupported.html) were added for some selected configurations.
  [(#2540)](https://github.com/PennyLaneAI/pennylane/pull/2540)

* Added a note for the [Depolarization Channel](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.DepolarizingChannel.html)
  that specifies how the channel behaves for the different values of depolarization probability `p`.
  [(#2669)](https://github.com/PennyLaneAI/pennylane/pull/2669)

* The quickstart documentation has been improved.
  [(#2530)](https://github.com/PennyLaneAI/pennylane/pull/2530)
  [(#2534)](https://github.com/PennyLaneAI/pennylane/pull/2534)
  [(#2564](https://github.com/PennyLaneAI/pennylane/pull/2564)
  [(#2565](https://github.com/PennyLaneAI/pennylane/pull/2565)
  [(#2566)](https://github.com/PennyLaneAI/pennylane/pull/2566)
  [(#2607)](https://github.com/PennyLaneAI/pennylane/pull/2607)
  [(#2608)](https://github.com/PennyLaneAI/pennylane/pull/2608)
  
* The quantum chemistry quickstart documentation has been improved.
  [(#2500)](https://github.com/PennyLaneAI/pennylane/pull/2500)
  

> * Improved testing documentation.

Author: ```suggestion
* Testing documentation has been improved.
```

  [(#2536)](https://github.com/PennyLaneAI/pennylane/pull/2536)
  

> * Added documentation for pre-commit.

Author: ```suggestion
* Documentation for pre-commit has been added.
```

  [(#2567)](https://github.com/PennyLaneAI/pennylane/pull/2567)
  

> * Updated documentation for draw control wires change.

Author: ```suggestion
* Documentation for draw control wires change has been updated.
```
